pow/src/test/java/tech/pegasys/artemis/pow/MinimumGenesisTimeBlockFinderTest.java

```diff
@@ -0,0 +1,264 @@
+
+    //    mockLatestCanonicalBlock(1000);
+    EthBlock.Block estimationBlock = mockBlockForEth1Provider("0xbf", 1000, 1000);
+
+    setMinGenesisTime(500);
+
+    // 1002 - 502 = 500, 500 / 5 = 100, the estimated genesis block number should be:
+    // 1000 - 100 = 900
+
+    mockBlockForEth1Provider("0x11", 900, 600);
+
+    // since the estimated block still had higher timestamp than min genesis, we should explore
+    // downwards
+
+    mockBlockForEth1Provider("0x08", 899, 510);
+
+    // since the second requested block still had higher timestamp than min genesis, we should
+    // explore downwards
+
+    mockBlockForEth1Provider("0x00", 898, 490);
+
+    // since the last requested block now had lower timestamp than min genesis, we should publish
+    // the block
+    // right before this as the first valid block
+
+    EthBlock.Block minGenesisTimeBlock =
+        minimumGenesisTimeBlockFinder.findMinGenesisTimeBlockInHistory(estimationBlock).join();
+
+    assertThatIsBlock(minGenesisTimeBlock, "0x08", 899, 510);
+  }
+
+  @Test
+  void minGenesisBlock_AboveEstimatedBlock() {
+    Constants.SECONDS_PER_ETH1_BLOCK = UnsignedLong.valueOf(5);
+
+    //    mockLatestCanonicalBlock(1000);
+    EthBlock.Block estimationBlock = mockBlockForEth1Provider("0xbf", 1000, 1000);
+
+    setMinGenesisTime(500);
+
+    // 1002 - 502 = 500, 500 / 5 = 100, the estimated genesis block number should be:
+    // 1000 - 100 = 900
+
+    mockBlockForEth1Provider("0x08", 900, 400);
+
+    // since the estimated block still had lower timestamp than min genesis, we should explore
+    // upwards
+
+    mockBlockForEth1Provider("0x08", 901, 450);
+
+    // since the second requested block still had lower timestamp than min genesis, we should
+    // explore upwards
+
+    mockBlockForEth1Provider("0x08", 902, 510);
+
+    // since the last requested block now had higher timestamp than min genesis, we should publish
+    // the block
+
+    EthBlock.Block minGenesisTimeBlock =
+        minimumGenesisTimeBlockFinder.findMinGenesisTimeBlockInHistory(estimationBlock).join();
+
+    assertThatIsBlock(minGenesisTimeBlock, "0x08", 902, 510);
+  }
+
+  @Test
+  void minGenesisBlock_EstimatedBlockIsTheValidBlock() {
+    Constants.SECONDS_PER_ETH1_BLOCK = UnsignedLong.valueOf(5);
+
+    //    mockLatestCanonicalBlock(1000);
+    EthBlock.Block estimationBlock = mockBlockForEth1Provider("0xbf", 1000, 1000);
+
+    setMinGenesisTime(502);
+
+    // 1002 - 502 = 500, 500 / 5 = 100, the estimated genesis block number should be:
+    // 1000 - 100 = 900
+
+    mockBlockForEth1Provider("0x08", 900, 500);
+
+    // since the genesis time calculated from the , we should publish the block
+
+    EthBlock.Block minGenesisTimeBlock =
+        minimumGenesisTimeBlockFinder.findMinGenesisTimeBlockInHistory(estimationBlock).join();
+
+    assertThatIsBlock(minGenesisTimeBlock, "0x08", 900, 500);
+  }
+
+  //  @Test
+  //  void waitForFirstValidBlock() {
```

Comment:
Why are these all commented out?

Reply:
They became redundant with the tests written in DepositProcessingControllerTest. I forgot to delete them, they're deleted now.